tests/graph/test_isomorphic.py

```diff
@@ -228,3 +228,84 @@ def test_same_degrees_non_isomorphic(self):
                 self.assertFalse(
                     retworkx.is_isomorphic(g_a, g_b, id_order=id_order)
                 )
+
+    def test_graph_vf2_mapping_identical(self):
+        graph = retworkx.generators.grid_graph(2, 2)
+        second_graph = retworkx.generators.grid_graph(2, 2)
+        mapping = retworkx.graph_vf2_mapping(graph, second_graph)
+        self.assertEqual(mapping, {0: 0, 1: 1, 2: 2, 3: 3})
+
+    def test_graph_vf2_mapping_identical_removals(self):
+        graph = retworkx.generators.path_graph(2)
+        second_graph = retworkx.generators.path_graph(4)
+        second_graph.remove_nodes_from([1, 2])
+        second_graph.add_edge(0, 3, None)
+        mapping = retworkx.graph_vf2_mapping(graph, second_graph)
+        self.assertEqual({0: 0, 1: 3}, mapping)
+
+    def test_graph_vf2_mapping_identical_removals_first(self):
+        second_graph = retworkx.generators.path_graph(2)
+        graph = retworkx.generators.path_graph(4)
+        graph.remove_nodes_from([1, 2])
+        graph.add_edge(0, 3, None)
+        mapping = retworkx.graph_vf2_mapping(
+            graph,
+            second_graph,
+        )
+        self.assertEqual({0: 0, 3: 1}, mapping)
+
+    def test_subgraph_vf2_mapping(self):
+        graph = retworkx.generators.grid_graph(10, 10)
+        second_graph = retworkx.generators.grid_graph(2, 2)
+        mapping = retworkx.graph_vf2_mapping(graph, second_graph, subgraph=True)
+        self.assertEqual(mapping, {0: 0, 1: 1, 10: 2, 11: 3})
+
+    def test_graph_vf2_mapping_identical_vf2pp(self):
+        graph = retworkx.generators.grid_graph(2, 2)
+        second_graph = retworkx.generators.grid_graph(2, 2)
+        mapping = retworkx.graph_vf2_mapping(
+            graph, second_graph, id_order=False
+        )
+        valid_mappings = [
+            {0: 0, 1: 1, 2: 2, 3: 3},
+            {0: 0, 1: 2, 2: 1, 3: 3},
+        ]
```

Comment:
I think there are more valid mappings since node `0`  of `graph` can be mapped to any node of `second_graph` and still obtained an isomorphism

Reply:
Yeah, there are a lot more valid mappings. I should rename this to `expected_mappings`. I basically just ran `tox -epy -- vf2 --until-failure` for like 500 iterations without failures to try and find an exhaustive set of mappings that this path could return. This isn't expected to be an exhaustive list of mappings, just those that I encountered when running the code. Unless you have a better suggestion on how to test this? Maybe just come up with an example that only has 1 possible mapping

Reply:
i don't think there are that many, it should be 8 in total so in principle we could exhaustively list them. this might become a problem in the following test with subgraphs. maybe use a path graph that has just two possible mappings?
 
but honestly i'm ok to leave it as it is.